#!/usr/bin/perl
use strict;
use warnings;

use CQS::ClassFactory;
use CQS::FileUtils;
use CQS::SystemUtils;
use CQS::ConfigUtils;
use Pipeline::RNASeq;

my $def = {

  #define task name, this name will be used as prefix of a few result, such as read count table file name.
  task_name => "RNAseq_mouse",

  #email which will be used for notification if you run through cluster
  email => "baohong.liu\@vanderbilt.edu",

  #target dir which will be automatically created and used to save code and result
  target_dir => "/scratch/liub11/Vanguard/09292017_llXu_RNAseq_mouse",

  #cqs in house software which will be used to generate count table. https://github.com/shengqh/CQS.Tools/
  cqstools => "/home/shengq2/cqstools/cqstools.exe",

  #gene annotation file
  transcript_gtf => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.gtf",

  #gene name to transcription name map file which is generated by "mono cqstools.exe gtf_buildmap".
  name_map_file => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.map",

  #genome sequence file
  fasta_file => "/scratch/cqs/shengq2/references/gencode/mm10/GRCm38.p5.genome.fa",

  #using STAR.
  aligner    => "star",
  star_index => "/scratch/cqs/shengq2/references/gencode/mm10/STAR_index_2.5.2b_gencodeVM12_sjdb99",

  #perform RNASeQC or not
  perform_rnaseqc => 0,
  rnaseqc_jar     => "/home/shengq2/local/bin/RNA-SeQC_v1.1.8.jar",

  #source files
  files => {
  "C57_Nor1" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-1_S1_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-1_S1_R2_001.fastq.gz"],
  "C57_Nor2" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-2_S2_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-2_S2_R2_001.fastq.gz"],
  "C57_Nor3" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-3_S3_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-3_S3_R2_001.fastq.gz"],
  "C57_Hy1" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-4_S4_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-4_S4_R2_001.fastq.gz"],
  "C57_Hy2" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-5_S5_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-5_S5_R2_001.fastq.gz"],
  "C57_Hy3" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-6_S6_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-6_S6_R2_001.fastq.gz"],
  "Perko_Nor1" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-7_S7_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-7_S7_R2_001.fastq.gz"],
  "Perko_Nor2" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-8_S8_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-8_S8_R2_001.fastq.gz"],
  "Perko_Nor3" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-9_S9_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-9_S9_R2_001.fastq.gz"],
  "Perko_Hy1" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-10_S10_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-10_S10_R2_001.fastq.gz"],
  "Perko_Hy2" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-11_S11_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-11_S11_R2_001.fastq.gz"],
  "Perko_Hy3" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-12_S12_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-12_S12_R2_001.fastq.gz"],
  "Bmal1ko_Nor1" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-13_S14_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-13_S14_R2_001.fastq.gz"],
  "Bmal1ko_Nor2" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-14_S15_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-14_S15_R2_001.fastq.gz"],
  "Bmal1ko_Nor3" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-15_S16_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-15_S16_R2_001.fastq.gz"],
  "Bmal1ko_Hy1" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-16_S17_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-16_S17_R2_001.fastq.gz"],
  "Bmal1ko_Hy2" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-17_S18_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-17_S18_R2_001.fastq.gz"],
  "Bmal1ko_Hy3" => ["/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-18_S19_R1_001.fastq.gz","/gpfs23/scratch/liub11/Vanguard/llXu_RawData/755-LX-18_S19_R2_001.fastq.gz"],
},

  #group information for visualization and comparison
  groups => {
    "C57_Nor" => [ "C57_Nor1","C57_Nor2","C57_Nor3" ],
    "C57_Hy"    => [ "C57_Hy1","C57_Hy2","C57_Hy3" ],
    "Perko_Nor" => [ "Perko_Nor1","Perko_Nor2","Perko_Nor3" ],
    "Perko_Hy" => [ "Perko_Hy1","Perko_Hy2","Perko_Hy3" ],
    "Bmal1ko_Nor" => [ "Bmal1ko_Nor1","Bmal1ko_Nor2","Bmal1ko_Nor3" ],
    "Bmal1ko_Hy" => [ "Bmal1ko_Hy1","Bmal1ko_Hy2","Bmal1ko_Hy3" ],
        
  },

  #Comparison information, in each comparison, the first one is control. For example, in comparison "DMSO_vs_FED", "FED" is control.
  pairs => {
    "C57_Hy_vs_C57_Nor" => [ "C57_Nor", "C57_Hy" ], 
   	"Perko_Hy_vs_Perko_Nor" => [ "Perko_Nor","Perko_Hy" ],	
    "Bmal1ko_Hy_vs_Bmal1ko_Nor" => [ "Bmal1ko_Nor","Bmal1ko_Hy" ],  
    "Perko_Nor_vs_C57_Nor" => [ "C57_Nor","Perko_Nor" ], 
    "Bmal1ko_Nor_vs_C57_Nor" => [ "C57_Nor","Bmal1ko_Nor" ],		
    "Bmal1ko_Nor_vs_Perko_Nor" => [ "Perko_Nor","Bmal1ko_Nor" ],	
    "Perko_Hy_vs_C57_Hy" => [ "C57_Hy","Perko_Hy" ],
    "Bmal1ko_Hy_vs_C57_Hy" => [ "C57_Hy","Bmal1ko_Hy" ],	
    "Bmal1ko_Hy_vs_Perko_Hy" => [ "Perko_Hy","Bmal1ko_Hy" ],	
    
  },
  
  #perform enrichment analysis, organism should be: "athaliana" "btaurus"       "celegans"      "cfamiliaris"   "drerio"  "sscrofa"       "dmelanogaster" "ggallus"       "hsapiens"      "mmusculus"  "rnorvegicus"   "scerevisiae"   
  perform_webgestalt => 1,
  webgestalt_organism => "mmusculus", 
  
  #perform multiqc
  perform_multiqc => 1,
};

performRNASeq($def);

1;
