#!/usr/bin/perl
use strict;
use warnings;

use CQS::ClassFactory;
use CQS::FileUtils;
use CQS::SystemUtils;
use CQS::ConfigUtils;
use Pipeline::RNASeq;

my $def = {

  #define task name, this name will be used as prefix of a few result, such as read count table file name.
  task_name => "RNAseq_human",

  #email which will be used for notification if you run through cluster
  email => "baohong.liu\@vanderbilt.edu",

  #target dir which will be automatically created and used to save code and result
  target_dir => "/gpfs23/scratch/VANGARD/20171127_Dylan_RNA_Seq_human_cell/project1/RNA_Seq1",

  #cqs in house software which will be used to generate count table. https://github.com/shengqh/CQS.Tools/
  cqstools => "/home/shengq2/cqstools/cqstools.exe",

#gene annotation file
  transcript_gtf => "/scratch/cqs/shengq2/references/gatk/b37/Homo_sapiens.GRCh37.82.MT.gtf",
#    
#gene name to transcription name map file which is generated by "mono cqstools.exe gtf_buildmap".
  name_map_file  => "/scratch/cqs/shengq2/references/gatk/b37/Homo_sapiens.GRCh37.82.MT.map",
      
# genome sequence file
  fasta_file     => "/scratch/cqs/shengq2/references/gatk/b37/human_g1k_v37.fasta",  

  #using STAR.
  aligner => "star",
  star_index => "/scratch/cqs/shengq2/references/gatk/b37/STAR_index_2.5.2b_ensembl_v82_sjdb99",  
  
#perform RNASeQC or not
  perform_rnaseqc => 0,
  rnaseqc_jar     => "/home/shengq2/local/bin/RNA-SeQC_v1.1.8.jar",

  #source files
  files => {
  "729_AF_7" => ["/scratch/VANGARD/20171127_Dylan_RNA_Seq_human_cell/project1/729-AF-7_S43_R1_001.fastq.gz","/scratch/VANGARD/20171127_Dylan_RNA_Seq_human_cell/project1/729-AF-7_S43_R2_001.fastq.gz"],
  "729_AF_8" => ["/scratch/VANGARD/20171127_Dylan_RNA_Seq_human_cell/project1/729-AF-8_S44_R1_001.fastq.gz","/scratch/VANGARD/20171127_Dylan_RNA_Seq_human_cell/project1/729-AF-8_S44_R2_001.fastq.gz"],
},

  #group information for visualization and comparison
  groups => {
    "group1" => [ "729_AF_7" ],
    "group2" => [ "729_AF_8" ],
        
  },

  #Comparison information, in each comparison, the first one is control. For example, in comparison "DMSO_vs_FED", "FED" is control.
  pairs => {
    "group2_vs_group1" => [ "group1", "group2" ], 
    		
    
  },
  
  #perform enrichment analysis, organism should be: "athaliana" "btaurus"       "celegans"      "cfamiliaris"   "drerio"  "sscrofa"       "dmelanogaster" "ggallus"       "hsapiens"      "mmusculus"  "rnorvegicus"   "scerevisiae"   
  perform_webgestalt => 1,
  webgestalt_organism => "hsapiens", 
  
  #perform multiqc
  perform_multiqc => 1,
};

performRNASeq($def);

1;
