#!/usr/bin/perl
use strict;
use warnings;

use CQS::ClassFactory;
use CQS::FileUtils;
use CQS::SystemUtils;
use CQS::ConfigUtils;
use Pipeline::RNASeq;

my $def = {

  #define task name, this name will be used as prefix of a few result, such as read count table file name.
  task_name => "RNAseq_mouse",

  #email which will be used for notification if you run through cluster
  email => "baohong.liu\@vanderbilt.edu",

  #target dir which will be automatically created and used to save code and result
  target_dir => "/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953_result",

  #cqs in house software which will be used to generate count table. https://github.com/shengqh/CQS.Tools/
  cqstools => "/home/shengq2/cqstools/cqstools.exe",

  #gene annotation file
  transcript_gtf => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.gtf",

  #gene name to transcription name map file which is generated by "mono cqstools.exe gtf_buildmap".
  name_map_file => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.map",

  #genome sequence file
  fasta_file => "/scratch/cqs/shengq2/references/gencode/mm10/GRCm38.p5.genome.fa",

  #using STAR.
  aligner    => "star",
  star_index => "/scratch/cqs/shengq2/references/gencode/mm10/STAR_index_2.5.2b_gencodeVM12_sjdb99",

  #perform RNASeQC or not
  perform_rnaseqc => 0,
  rnaseqc_jar     => "/home/shengq2/local/bin/RNA-SeQC_v1.1.8.jar",

  #source files
  files => {
  "953-AQ-1" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-1_S414_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-1_S414_R2_001.fastq.gz"],
  "953-AQ-2" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-2_S415_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-2_S415_R2_001.fastq.gz"],
  "953-AQ-3" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-3_S416_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-3_S416_R2_001.fastq.gz"],
  "953-AQ-4" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-4_S417_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-4_S417_R2_001.fastq.gz"],
  "953-AQ-5" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-5_S418_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-5_S418_R2_001.fastq.gz"],
  "953-AQ-6" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-6_S419_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-6_S419_R2_001.fastq.gz"],
  "953-AQ-7" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-7_S420_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-7_S420_R2_001.fastq.gz"],
  "953-AQ-8" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-8_S421_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-8_S421_R2_001.fastq.gz"],
  "953-AQ-9" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-9_S422_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-9_S422_R2_001.fastq.gz"],
  "953-AQ-10" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-10_S423_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-10_S423_R2_001.fastq.gz"],
  "953-AQ-11" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-11_S424_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-11_S424_R2_001.fastq.gz"],
  "953-AQ-12" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-12_S425_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-12_S425_R2_001.fastq.gz"],
  "953-AQ-13" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-13_S426_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-13_S426_R2_001.fastq.gz"],
  "953-AQ-14" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-14_S427_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-14_S427_R2_001.fastq.gz"],
  "953-AQ-15" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-15_S428_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-15_S428_R2_001.fastq.gz"],
  "953-AQ-16" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-16_S429_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-16_S429_R2_001.fastq.gz"],
  "953-AQ-17" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-17_S430_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-17_S430_R2_001.fastq.gz"],
  "953-AQ-18" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-18_S431_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-18_S431_R2_001.fastq.gz"],
  "953-AQ-19" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-19_S432_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-19_S432_R2_001.fastq.gz"],
  "953-AQ-20" => ["/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-20_S433_R1_001.fastq.gz","/gpfs23/scratch/VANGARD/20171117_Aidong_RNA_seq_mouse/953/953-AQ-20_S433_R2_001.fastq.gz"],
},

  #group information for visualization and comparison
  groups => {
    "group1" => [ "953-AQ-1","953-AQ-2","953-AQ-3","953-AQ-4","953-AQ-5" ],
    "group2" => [ "953-AQ-6","953-AQ-7","953-AQ-8","953-AQ-9","953-AQ-10" ],
    "group3" => [ "953-AQ-11","953-AQ-12","953-AQ-13","953-AQ-14","953-AQ-15" ],
    "group4" => [ "953-AQ-16","953-AQ-17","953-AQ-18","953-AQ-19","953-AQ-20" ],
        
  },

  #Comparison information, in each comparison, the first one is control. For example, in comparison "DMSO_vs_FED", "FED" is control.
  pairs => {
    "group2_vs_group1" => [ "group1", "group2" ], 
   "group3_vs_group1" => [ "group1","group3" ],	
    "group4_vs_group1" => [ "group1","group4" ],  
    "group4_vs_group3" => [ "group3","group4" ], 
    		
    
    	
    
  },
  
  #perform enrichment analysis, organism should be: "athaliana" "btaurus"       "celegans"      "cfamiliaris"   "drerio"  "sscrofa"       "dmelanogaster" "ggallus"       "hsapiens"      "mmusculus"  "rnorvegicus"   "scerevisiae"   
  perform_webgestalt => 1,
  webgestalt_organism => "mmusculus", 
  
  #perform multiqc
  perform_multiqc => 1,
};

performRNASeq($def);

1;
