#!/usr/bin/perl
use strict;
use warnings;

use CQS::ClassFactory;
use CQS::FileUtils;
use CQS::SystemUtils;
use CQS::ConfigUtils;
use CQS::PerformRNAseq;

my $def = {

  #define task name, this name will be used as prefix of a few result, such as read count table file name.
  task_name => "RNAseq_BTaurus",

  #email which will be used for notification if you run through cluster
  email => "marisol.a.ramirez\@vanderbilt.edu",

  #target dir which will be automatically created and used to save code and result
  target_dir => "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/result_UMD",

#cqs in house software which will be used to generate count table. https://github.com/shengqh/CQS.Tools/
 cqstools => "/home/shengq2/cqstools/cqstools.exe",

 #gene annotation file

 #gene name to transcription name map file which is generated by "mono cqstools.exe gtf_buildmap".

 #genome sequence file

 #using STAR.
  aligner    => "star",

 #perform RNASeQC or not
  perform_rnaseqc => 0,
  rnaseqc_jar     => "/home/shengq2/local/bin/RNA-SeQC_v1.1.8.jar",
 
 #preprocesing
pairend => 1, 
                         
  #source files
files => {
  "HD_01" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-1-CGGCTATG-GGCTCTGA_S18_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-1-CGGCTATG-GGCTCTGA_S18_R2_001.fastq.gz"],
  "HD_02" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-2-TCCGCGAA-GGCTCTGA_S19_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-2-TCCGCGAA-GGCTCTGA_S19_R2_001.fastq.gz"],
  "HD_03" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-3-TCTCGCGC-GGCTCTGA_S20_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-3-TCTCGCGC-GGCTCTGA_S20_R2_001.fastq.gz"],
  "HD_04" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-4-AGCGATAG-GGCTCTGA_S21_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-4-AGCGATAG-GGCTCTGA_S21_R2_001.fastq.gz"],
  "HD_05" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-5-ATTACTCG-AGGCGAAG_S22_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-5-ATTACTCG-AGGCGAAG_S22_R2_001.fastq.gz"],
  "HD_06" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-6-TCCGGAGA-AGGCGAAG_S23_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-6-TCCGGAGA-AGGCGAAG_S23_R2_001.fastq.gz"],
  "HD_07" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-7-CGCTCATT-AGGCGAAG_S24_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-7-CGCTCATT-AGGCGAAG_S24_R2_001.fastq.gz"],
  "HD_08" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-8-GAGATTCC-AGGCGAAG_S25_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-8-GAGATTCC-AGGCGAAG_S25_R2_001.fastq.gz"],
  "HD_09" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-9-ATTCAGAA-AGGCGAAG_S26_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-9-ATTCAGAA-AGGCGAAG_S26_R2_001.fastq.gz"],
  "HD_10" => ["/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-10-GAATTCGT-AGGCGAAG_S27_R1_001.fastq.gz", "/scratch/VANGARD/20180313_1024_HeatherDunn_BosTaurus_RNAseq/1024-HD-10-GAATTCGT-AGGCGAAG_S27_R2_001.fastq.gz"],
},


  #group information for visualization and comparison

  #Comparison information, in each comparison, the first one is control. For example, in comparison "DMSO_vs_FED", "FED" is control.

#perform enrichment analysis, organism should be: "athaliana" "btaurus"       "celegans"      "cfamiliaris"   "drerio"  "sscrofa"       "dmelanogaster" "ggallus"       "hsapiens"      "mmusculus"  "rnorvegicus"   "scerevisiae"

#perform multiqc
 perform_multiqc => 1,

#perfomr report
perform_report => 1,
        };


 performRNASeq_ncbi_UMD311($def);

 1;
