#!/usr/bin/perl
use strict;
use warnings;

use CQS::ClassFactory;
use CQS::FileUtils;
use CQS::SystemUtils;
use CQS::ConfigUtils;
use CQS::PerformRNAseq;

my $def = {

  #define task name, this name will be used as prefix of a few result, such as read count table file name.
  task_name => "RNAseq_mouse",

  #email which will be used for notification if you run through cluster
  email => "marisol.a.ramirez\@vanderbilt.edu",

  #target dir which will be automatically created and used to save code and result
  target_dir => "/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/result_cutadapt",

#cqs in house software which will be used to generate count table. https://github.com/shengqh/CQS.Tools/
 cqstools => "/home/shengq2/cqstools/cqstools.exe",

 #gene annotation file
  transcript_gtf => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.gtf",

 #gene name to transcription name map file which is generated by "mono cqstools.exe gtf_buildmap".
  name_map_file => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.map",

 #genome sequence file
  fasta_file => "/scratch/cqs/shengq2/references/gencode/mm10/GRCm38.p5.genome.fa",

 #using STAR.
  aligner    => "star",
  star_index => "/scratch/cqs/shengq2/references/gencode/mm10/STAR_index_2.5.2b_gencodeVM12_sjdb99",

 #perform RNASeQC or not
  perform_rnaseqc => 0,
  rnaseqc_jar     => "/home/shengq2/local/bin/RNA-SeQC_v1.1.8.jar",

  #perform cutadapt
  perform_cutadapt => 1,
  cutadapt_option  => "-O 1",
  adapter          => "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC",    #trueseq adapter
  adapter_5 => "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA",
  min_read_length  => 30,

                         
  #source files
files => {
  "TOM_FACS_neg_1" => ["/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-15-ACTTGATG-AGGCGAAG_S26_R1_001.fastq.gz", "/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-15-ACTTGATG-AGGCGAAG_S26_R2_001.fastq.gz"],
  "TOM_FACS_neg_2" => ["/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-16-GATCAGCG-AGGCGAAG_S27_R1_001.fastq.gz", "/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-16-GATCAGCG-AGGCGAAG_S27_R2_001.fastq.gz"],
  "TOM_FACS_pos_1" => ["/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-17-TAGCTTGT-AGGCGAAG_S28_R1_001.fastq.gz", "/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-17-TAGCTTGT-AGGCGAAG_S28_R2_001.fastq.gz"],
  "TOM_FACS_pos_2" => ["/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-18-GGCTACAG-AGGCGAAG_S29_R1_001.fastq.gz", "/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-18-GGCTACAG-AGGCGAAG_S29_R2_001.fastq.gz"],
  "TOM_FACS_pos_3" => ["/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-19-CTTGTACT-AGGCGAAG_S30_R1_001.fastq.gz", "/scratch/VANGARD/20180425_1478_Levine_RNAseq_mouse/raw_1478/1478-MR-19-CTTGTACT-AGGCGAAG_S30_R2_001.fastq.gz"],
},

  
  #group information for visualization and comparison
  groups => {
    "TOM_FACS_neg" => ["TOM_FACS_neg_1","TOM_FACS_neg_2" ],
    "TOM_FACS_pos" => ["TOM_FACS_pos_1","TOM_FACS_pos_2", "TOM_FACS_pos_3" ],
 
  },

  #Comparison information, in each comparison, the first one is control. For example, in comparison "DMSO_vs_FED", "FED" is control.
  pairs => {
   "TOM_FACS_pos_vs_TOM_FACS_pos" => ["TOM_FACS_neg","TOM_FACS_pos" ],

  },

#perform enrichment analysis, organism should be: "athaliana" "btaurus"       "celegans"      "cfamiliaris"   "drerio"  "sscrofa"       "dmelanogaster" "ggallus"       "hsapiens"      "mmusculus"  "rnorvegicus"   "scerevisiae"
  perform_webgestalt => 1,
    webgestalt_organism => "mmusculus",

#perform multiqc
        perform_multiqc => 1,

#perfomr report
perform_report =>1 ,

        };


performRNASeq_gencode_mm10($def);

 1;
