#!/usr/bin/perl
use strict;
use warnings;

use CQS::ClassFactory;
use CQS::FileUtils;
use CQS::SystemUtils;
use CQS::ConfigUtils;
use CQS::PerformRNAseq;

my $def = {

  #define task name, this name will be used as prefix of a few result, such as read count table file name.
  task_name => "RNAseq_mouse",

  #email which will be used for notification if you run through cluster
  email => "marisol.a.ramirez\@vanderbilt.edu",

  #target dir which will be automatically created and used to save code and result
  target_dir => "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/20180226_resultcutadapt",

#cqs in house software which will be used to generate count table. https://github.com/shengqh/CQS.Tools/
 cqstools => "/home/shengq2/cqstools/cqstools.exe",

 #gene annotation file
  transcript_gtf => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.gtf",

 #gene name to transcription name map file which is generated by "mono cqstools.exe gtf_buildmap".
  name_map_file => "/scratch/cqs/shengq2/references/gencode/mm10/gencode.vM12.chr_patch_hapl_scaff.annotation.map",

 #genome sequence file
  fasta_file => "/scratch/cqs/shengq2/references/gencode/mm10/GRCm38.p5.genome.fa",

 #using STAR.
  aligner    => "star",
  star_index => "/scratch/cqs/shengq2/references/gencode/mm10/STAR_index_2.5.2b_gencodeVM12_sjdb99",

 #perform RNASeQC or not
  perform_rnaseqc => 0,
  rnaseqc_jar     => "/home/shengq2/local/bin/RNA-SeQC_v1.1.8.jar",
 
 #preprocesing
  run_cutadapt  => 1,
  adapter      => "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC",
  perform_cutadapt => 1,
  min_read_length => 30,
 
                         
  #source files
  files => { 
  "LRY-01_WTa" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-1_S42_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-1_S42_R2_001.fastq.gz"],
  "LRY-02_WTb" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-2_S43_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-2_S43_R2_001.fastq.gz"],
  "LRY-03_WTc" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-3_S44_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-3_S44_R2_001.fastq.gz"],
  "LRY-04_WTd" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-4_S45_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-4_S45_R2_001.fastq.gz"],
  "LRY-05_HPS2a" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-5_S46_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-5_S46_R2_001.fastq.gz"],
  "LRY-06_HPS2b" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-6_S47_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-6_S47_R2_001.fastq.gz"],
  "LRY-07_HPS2c" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-7_S48_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-7_S48_R2_001.fastq.gz"],
  "LRY-09_HPS3a" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-9_S49_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-9_S49_R2_001.fastq.gz"],
  "LRY-10_HPS3b" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-10_S50_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-10_S50_R2_001.fastq.gz"],
  "LRY-11_HPS3c" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-11_S51_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-11_S51_R2_001.fastq.gz"],
  "LRY-12_HPS3d" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-12_S52_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-12_S52_R2_001.fastq.gz"],
  "LRY-13_HPS2e" => ["/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-13-GTTTCGGA-TCTTTCCC_S48_R1_001.fastq.gz", "/scratch/VANGARD/20171219_Aidong_RNA_seq_mouse/3700/3700-LRY-13-GTTTCGGA-TCTTTCCC_S48_R2_001.fastq.gz"],

  
},

  #group information for visualization and comparison
  groups => {
    "WT" => ["LRY-01_WTa", "LRY-02_WTb", "LRY-03_WTc" , "LRY-04_WTd"],
    "HPS2" => [ "LRY-05_HPS2a",  "LRY-06_HPS2b","LRY-07_HPS2c", "LRY-13_HPS2e"],
    "HPS3" => [ "LRY-09_HPS3a", "LRY-10_HPS3b","LRY-11_HPS3c", "LRY-12_HPS3d" ],
    "HPS3_WT" => ["LRY-01_WTa","LRY-02_WTb","LRY-03_WTc" ,"LRY-04_WTd","LRY-09_HPS3a", 
"LRY-10_HPS3b","LRY-11_HPS3c", "LRY-12_HPS3d" ],
  },

  #Comparison information, in each comparison, the first one is control. For example, in comparison "DMSO_vs_FED", "FED" is control.
  pairs => {

"HPS2_vs_HPS3" => ["HPS3", "HPS2"],
"HPS2_vs_HPS3_WT" => ["HPS3_WT", "HPS2"],
"HPS2_vs_WT" => ["WT", "HPS2"],
"HPS3_vs_WT" => ["WT", "HPS3"],
  },

#perform enrichment analysis, organism should be: "athaliana" "btaurus"       "celegans"      "cfamiliaris"   "drerio"  "sscrofa"       "dmelanogaster" "ggallus"       "hsapiens"      "mmusculus"  "rnorvegicus"   "scerevisiae"
  perform_webgestalt => 1,
    webgestalt_organism => "mmusculus",

#perform multiqc
        perform_multiqc => 1,

#perfomr report
perform_report => 1,

        };


 performRNASeq_gencode_mm10($def);

 1;
